images/istio/tests/main.tf

```diff
@@ -96,16 +95,19 @@ resource "helm_release" "istiod" {
     }
     global = {
       istioNamespace = local.namespace
-      # We have to trim the suffix and specify it in `image`, because this
-      # Helm chart does not like slashes in the image name.
-      hub = replace(data.oci_string.proxy-ref.registry_repo, "/istio-proxy", "")
-      tag = data.oci_string.proxy-ref.pseudo_tag
+      # These Helm charts do not like slashes in the image name.
+      #
+      # If the registry_repo is gcr.io/my/repo/istio-proxy, we need to set
+      #   hub   = gcr.io/my/repo
+      #   image = istio-proxy
+      hub = replace(data.oci_string.proxy-ref.registry_repo, "//[^/]*$/", "")
       proxy = {
-        image = "istio-proxy"
+        image = replace(data.oci_string.proxy-ref.registry_repo, "/^.*//", "")
```

Comment:
This pattern seems wrong 🤔

Reply:
changed to use `basename` / `dirname` now.

Reply:
(the pattern is weird because `replace` only treat the search string as regexp if it begins and ends w/ slashes -- very confusing)